### DISCUSSION TITLE
Antireflection foil/ Entspiegelungsfolie

### DISCUSSION THREAD

**Discussion by Gunter1710** [jomjol/AI-on-the-edge-device#651] [Q&A]
My gas meter reflects very strongly. Does anyone have experience with reflective foil?

Mein Gaszähler reflektiert sehr stark. Hat jemand Erfahrungen mit Entspieglungsfolie?

Regards
Gunter1710

**Comment by dev8-de** [comment]
Interssante Idee, hab mir mal ne einfache Smartphone-Entspiegelungsfolie bestellt, mal gucken ob das was bringt...

**Reply by caco3** [reply]
Backpapier als Diffusor vor die LED (mit etwas Abstand) hat bei mir auch einiges geholfen.

**Comment by dev8-de** [comment]
Hatte ich probert, nicht wirklich viel geholfen. Ich habe jetzt schwarzen Schaumstoff dazwischen aber auch das erzeugt noch Refektionen... 

So sieht das bei mir aus:

![gas](https://user-images.githubusercontent.com/11597226/195707370-134c0bc9-b4b3-4583-b844-134d777c3d9f.jpg)

Nachtrag: Auch ist das schon gewinkelt, also nicht direkt drauf.

**Reply by caco3** [reply]
So schlecht sieht es doch gar nicht aus!

Könnte mattes Klebeband direkt auf die Digits helfen?

Oder den ESP drehen, dass die LED eine andere Position hat.

Oder eine externe LED.

**Reply by gfehr** [reply]
Bei mir hat es geholfen, vor der LED einen Klebestreifen (weiß, 5 mm Abstand)  anzubringen. Der Kamera Mittelpunkt ist dann noch 2 cm über der Anzeige. Dafür konnte der Winkel parallel zur Anzeige bleiben denn bei einer Verkippung gab es vermehrt Erkennungsfehler.

**Reply by dev8-de** [reply]
Siehe: https://github.com/jomjol/AI-on-the-edge-device/discussions/651#discussioncomment-3947711

**Comment by dev8-de** [comment]
Ja, aber die Reflexionen auf den letzten beiden Stellen vor dem Komma sorgen bei mir für Erkennungsfehler. 1 wird zu 7, 8 zu ner 3 und umgekehrt. Da haste dann mit mal sowas:

{'value': '', 'raw': '35871.002', 'error': 'Rate too high - Read: 35871.002 - Pre: 35810.962', 'rate': '', 'timestamp': '2022-10-14T08:35:03'}

Und hab schon schwarzen Schaumstoff vor der LED geklebt und angewinkelt ist der auch, d.h. die LED strahlt nicht direkt drauf.. Daher probier ich das jetzt mit Anti-Reflex-Folie für Smartphones. Kommt Montag, ich berichte. :)

**Comment by dev8-de** [comment]
Moin,

also bei mir hat es nichts gebracht, das ganze Bild wird dadurch matschig und nicht mehr wirklich erkenntbar.

Gruß.

**Reply by caco3** [reply]
hast du sie auf das meter geklebt? falls ja, hast du ausschnitte für die digits gemacht?
ich verwende backpapier als Diffusor, aber natürlich nur über dem Bereich der LEDs

**Reply by dev8-de** [reply]
Ne, ich hatte das komplett quer über alle Zahlen geklebt.. Aber ich habe mir mittlerweile ein Gehäuse dafür gedruckt (war vorher innem Pappkarton geklebt) und durch nen Berechnungsfehler ist die LED nun leicht verdeckt, was ausgereicht hat, damit die nicht mehr so draufknallt und die Zahlen werden nun korrekt erkannt...

![gaszähler](https://user-images.githubusercontent.com/11597226/197461996-336bb78e-675b-4056-a5c9-bac0622ec088.jpg)

Jedenfalls musste mein Korrekurscript nur noch bei dem Rollover vom Wasserzähler eingreifen...

**Comment by ozett** [comment]
to reduce reflections: 

maybe a ping-pong ball as deflektor?
https://hackaday.com/2022/01/04/modular-design-enables-huge-ping-pong-ball-led-displays/

or external led on GPIo12 ?
https://github.com/jomjol/AI-on-the-edge-device/wiki/External-LED

**Comment by faberlau** [comment]
I have tried several different options. They all did not really help. The only thing that helped is: the light output was brought closer to the surface of the meter by a light channel. The effect is comparable to an external LED mounted on the side.

My setup on top of water counter:
![img2](https://user-images.githubusercontent.com/111505155/211215017-b10a91c9-ce97-41c9-9d2d-a1f3920edfbc.jpg) 

The licht channel:
![img1](https://user-images.githubusercontent.com/111505155/211215025-96cced5a-450e-44ba-9fc5-f19edaa74a47.jpg)

The result:
![img5](https://user-images.githubusercontent.com/111505155/211215043-73a9a47c-7828-4f05-8dc1-7292834d85c4.jpg)

**Reply by leergut** [reply]
Nice! How did you built the light channel? Is it made with semi light-transmissive paper? like frost foil?

**Reply by faberlau** [reply]
I made the light channel from white corrugated cardboard. It is glued to the inside of the housing with adhesive tape.

**Comment by ozett** [comment]
like a diffusor... great.  https://letmegooglethat.com/?q=flashlight+diffusor

i am struggling with the same problem on my gasmeter.
may i try some diffusor variants. 

your image from the counter looks clear!  great!
